src/raylib.h

```diff
@@ -1177,6 +1177,7 @@ RLAPI bool IsKeyReleased(int key);                            // Check if a key
 RLAPI bool IsKeyUp(int key);                                  // Check if a key is NOT being pressed
 RLAPI int GetKeyPressed(void);                                // Get key pressed (keycode), call it multiple times for keys queued, returns 0 when the queue is empty
 RLAPI int GetCharPressed(void);                               // Get char pressed (unicode), call it multiple times for chars queued, returns 0 when the queue is empty
+RLAPI int GetKeyPressedPro(int* scanCode);                       // Get key pressed (keycode), and optional scancode, call it multiple times for keys queued, returns 0 when the queue is empty
```

Comment:
As per raylib design, I try to minimize return-pointers when possible. I prefer a ` int GetKeyPressedRaw(void)` function to note the raw nature (scancode) of returned key.

Reply:
The problem with making a function that just returns the scan code is that this pulls from the events. Calling that function will eat the event and prevent you from beeing able to get the Keyboard Key too.
There may be cases where the user wants both (such as for a "press a key" control config dialog), they need the KeyboardKey to use later for polling, but the scancode to be able to show the mapped key name.

The other option is to return the KeyInfo structure by value.